File: src/csharp/BelgianBeers/BelgianBeers/Tests/D01_LoadJson_Approach1.cs
using System.Linq;
using BelgianBeers.Models;
using BelgianBeers.Repositories;
using BelgianBeers.Tests.Utilities;
using Xunit;

namespace BelgianBeers.Tests
{
    [Collection("Demo 1 - Load JSON")]
    public class D01_LoadJson_Approach1
    {
        [Fact]
        public void LoadsDataFromJsonFile()
        {
            var repository = new BeersRepository();

            foreach (var (beerName, breweryName, rating, votes) in TestData.Beers)
            {
                // TODO DEMO: The get/add pattern is stupid! Add GetHashCode to Brewery() and just call .Add on the underlying hashset

                // Store the brewery
                var brewery = repository.GetBrewery(breweryName);
                if (brewery == null)
                {
                    brewery = new Brewery(breweryName);
                    repository.AddBrewery(brewery);
                }

                // Store the beer
                var beer = repository.GetBeer(beerName);
                if (beer == null)
                {
                    beer = new Beer(beerName, brewery, rating, votes);
                    repository.AddBeer(beer);
                }
            }

            Assert.True(repository.GetBeers().Any());
        }
    }
}

File: src/kotlin/BelgianBeers/src/test/java/org/jetbrains/kotlincsharpdemo/D01_LoadJson_Approach1.kt
package org.jetbrains.kotlincsharpdemo

import org.junit.Assert
import org.junit.Test

class D01_LoadJson_Approach1 {

  @Test
  fun load() {
    val breweries = mutableMapOf<Brewery, Brewery>()
    val beers = LinkedHashSet<Beer>()

    for ((beerName, breweryName, rating, votes) in TestData.beers) {
      if (beerName == null) continue

      val brewery = if (breweryName == null) {
        null
      } else {
        val b = Brewery(breweryName)
        breweries.getOrPut(b) { b }
      }

      beers += Beer(beerName, brewery, rating ?: 0.0, votes ?: 0.0)
    }

    Assert.assertTrue(beers.any())
  }
}
